src/main/java/fi/aalto/cs/apluscourses/intellij/model/CourseProject.java

```diff
@@ -45,6 +44,8 @@
 
   @NotNull
   public final Event exercisesUpdated;
+  
+  private volatile User user;
```

Comment:
I'm not sure if `volatile` is needed here, all methods that access this field are called from EDT. I guess it doesn't hurt, but is there a reason for it? @OlliKiljunen is it necessary here?

Reply:
That's changed in bannertime

Reply:
You're really proud of that branch name

Reply:
Yep, there are problems related to this field but they should be solved in the "banner" PR. `volatile` certainly is not the right choice here because it doesn't help with code like this
```java
if (user != null) {
  user.doSomething();
}
```
`user` can be changed between the null-check and the statement, so this can throw null exception.

> all methods that access this field are called from EDT

Methods `getUsername()` and `getAuthentication()` too? If so, they should be marked with `@RequiresEdt`. I wouldn't expect getters like that being thread-unsafe.

Reply:
> You're really proud of that branch name

At least he should be.